ee/modules/140-user-authz/images/webhook/web/hook/handler.go

```diff
@@ -313,14 +356,44 @@ func (h *Handler) affectedDirs(r *WebhookRequest) []DirectoryEntry {
 	return dirEntriesAffected
 }
 
+// getsthe map of namespaces matching specific label selector from k8s api client
+func (h *Handler) getNamespacesByLabelSelector(namespaceSelector NamespaceSelector) (map[string]struct{}, error) {
```

Comment:
Do we need to get all namespaces by the label selector for every request? This could be more effective. Even if you know that namespaces by the following label selector aren't changed, you still request the complete list from Kubernetes. If you have a selector for 25 namespaces, you constantly ask them from Kubernetes.

What can be done here?
1. Simple way: request only the namespace from the request and check its label. Use `Get` instead of `List`. Then check whether it matches label selectors or not on our side.
2. Harder way: using shared informer. There may be issues with informer synchronization and authorization requests, the case when a user creates a namespace and starts working with it immediately, e.g., with a helm chart.